### PR TITLE
Bug 1025937 - Silently drop null intents in HealthReportBroadcastService.

### PR DESCRIPTION
**Pull request by rnewman**
It looks like an IntentService can sometimes be restarted with a null intent. The right thing to do appears to be to ignore it — we'll get another intent if we have to do some work. (If we were restarted mid-operation, we just wait until the next time.)

### CODE REVIEW

**Comment by mcomella** [comment]
There appears to be `onHandleIntent` calls in `fxa.receivers.FxAccountDeletedService` and `sync.receivers.SyncAccountDeletedService` - is there any reason you didn't add the fix there as well?

File: src/main/java/org/mozilla/gecko/background/healthreport/HealthReportBroadcastService.java [hunk 1]
```diff
@@ -113,6 +118,11 @@ protected void onHandleIntent(Intent intent) {
    * Attempts to handle the given intent for FHR document upload. If it cannot, false is returned.
    */
   protected boolean attemptHandleIntentForUpload(final Intent intent) {
+    if (intent == null) {
```
Comment (mcomella) [inline src/main/java/org/mozilla/gecko/background/healthreport/HealthReportBroadcastService.java]: Why did you only do this for `HealthReportBroadcastService`? `AnnouncementsBroadcastService` uses a similar paradigm.

Comment (rnewman) [inline src/main/java/org/mozilla/gecko/background/healthreport/HealthReportBroadcastService.java]: I spent the most time in that file.

Updated push coming momentarily.

**Comment by rnewman** [comment]
This original push only touched descendents of `BackgroundService`, which the deleted services are not. Updated push coming momentarily.

**Comment by rnewman** [comment]
Comments addressed and merged. Thanks!

**Pull request closed without merging**